wtransport/src/config.rs

```diff
@@ -388,6 +388,103 @@ impl ServerConfigBuilder<states::WantsIdentity> {
             migration: true,
         })
     }
+
+    /// Configures the server with a custom transport configuration and a default TLS setup
+    /// using the provided [`Identity`].
+    ///
+    /// This method is useful for scenarios where you need to customize the transport settings
+    /// while relying on a default TLS configuration built from an [`Identity`]. It gives you
+    /// control over the transport layer while maintaining safe and standard TLS settings.
+    ///
+    /// # Parameters
+    ///
+    /// - `identity`: A reference to an [`Identity`] that contains the server's certificate and
+    ///   private key. This will be used to generate the default TLS configuration.
+    /// - `transport_config`: A custom [`TransportConfig`] instance that allows you to specify
+    ///   various transport-layer settings according to your requirements.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use wtransport::{Identity, ServerConfig, quinn::TransportConfig};
+    ///
+    /// // Load the server identity (certificate and private key)
+    /// let identity = Identity::self_signed(["localhost", "127.0.0.1", "::1"]).unwrap();
+    ///
+    /// // Create a custom TransportConfig with specific settings
+    /// let mut custom_transport_config = TransportConfig::default();
+    /// custom_transport_config.datagram_send_buffer_size(1024);
+    ///
+    /// // Create a ServerConfigBuilder with the custom transport configuration and default TLS settings
+    /// let server_config = ServerConfig::builder()
+    ///     .with_bind_default(4433)
+    ///     .with_custom_transport(&identity, custom_transport_config)
+    ///     .build();
+    /// ```
+    pub fn with_custom_transport(
+        self,
+        identity: &Identity,
+        transport_config: TransportConfig,
```

Comment:
As made for `rustls`. Please re-export this third party symbol. That is:

at the begging of the module, appending existing type aliases
```
/// Alias of [`crate::quinn::TransportConfig`].
pub type QuicTransportConfig = crate::quinn::TransportConfig;
```